### ISSUE TITLE
Class identifier not properly scoped in class expression

### ISSUE THREAD

**Issue by jleyba** [google/closure-compiler#1546]
``` javascript
var Clazz = class Foo {};
class Foo {}
```

For the above, the first `Foo` should only be visible within the class expression. The compiler appears to be pulling this out during transpilation, producing

```
ERROR - Illegal redeclared variable: Foo
class Foo {}
      ^

1 error(s), 0 warning(s)
```

This should be handled similarly to function declaration/expressions:

http://closure-compiler-debugger.appspot.com/#input0%3Dvar%2520Clazz%2520%253D%2520class%2520Foo%2520%257B%257D%253B%250Aclass%2520Foo%2520%257B%257D%250A%250Avar%2520one%2520%253D%2520function%2520two()%2520%257B%257D%253B%250Afunction%2520two()%2520%257B%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by Dominator008** [comment]
Thanks for the report. The error you are seeing should be fixed by #1551, but we will also need to change our transpilation to support referring to the class name in the methods.

**Comment by brad4d** [comment]
yep, I believe I fixed it and failed to realize this issue could be closed.
Thanks for pointing this out @teppeis
